src/schemes/boundary/open_boundary/system.jl

```diff
@@ -305,7 +305,10 @@ end
 
     # Check if particle is in- or outside the fluid domain.
     # `face_normal` is always pointing into the fluid domain.
```

Comment:
```suggestion
    # `face_normal` is always pointing into the fluid domain.
    # Since this function is called for a particle that left the boundary zone,
    # it is sufficient to check if it is closer to the free surface or the transition phase
    # to determine if it exited the boundary zone into the fluid or through the free surface.
```